### PR TITLE
[core] Port GitHub workflow for ensuring triage label is present

### PR DESCRIPTION
**Pull request by DanailH**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

For context: https://www.notion.so/mui-org/GitHub-Action-to-add-need-triage-label-if-none-is-present-e6357dacba7947a1879d684c7bb63939?d=4912f8bad6af466783a3ef21ac71bad2

### CODE REVIEW

File: .github/workflows/ensure-triage-label.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Ensure triage label is present
+on:
```
Comment (Janpot) [inline .github/workflows/ensure-triage-label.yml:2]: 🤔  I'm just realizing that a small improvement we perhaps could do is to trigger it also on label deletion https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#label. To make sure an issue always has at least this label

```yml
  label:
    types:
      - deleted
```

File: .github/workflows/ensure-triage-label.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - uses: actions/github-script@d7906e4ad0b1822421a7e6a35d5ca353c962f410 # v6
+        with:
+          script: |
+            const { data: labels } = await github.rest.issues.listLabelsOnIssue({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+            });
+
+            console.log(JSON.stringify(labels, null, 2))
```
Comment (Janpot) [inline .github/workflows/ensure-triage-label.yml:21]: 🙂 Accidental debug log. We can keep it though.

**Review by Janpot — APPROVED** [review]
Approving. There's a note about an enhancement, but non-blocking

File: .github/workflows/ensure-triage-label.yml (hunk 1 above)
Comment (DanailH) [reply .github/workflows/ensure-triage-label.yml:2]: Added the trigger

File: .github/workflows/ensure-triage-label.yml (hunk 2 above)
Comment (DanailH) [reply .github/workflows/ensure-triage-label.yml]: Removed

**Comment by oliviertassinari** [comment]
I made a small enhancement 68b4da1495774669138213fdf681f8a9ae6e712a in X, Core, Toolpad, solving https://github.com/mui/mui-x/security/code-scanning/22

**Comment by oliviertassinari** [comment]
Same logic added to all the repositories:

- https://github.com/mui/mui-private/commit/a54dd0eeea8da459b24309cb4ea1210dfcfc4737
- https://github.com/mui/mui-public/commit/f955f9a6baf86c721462a9ab82e53d17fcee4f64
- https://github.com/mui/mui-design-kits/commit/51a002bd471893ddbaff950f9a0e4b68e4af35d1

**Pull request merged**